### PR TITLE
refactor: [M3-6893] - MUI v5 Migration - `SRC > Components > Paper`

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Updates imports/exports of our `<Paper />` component to match our moden patterns

## Major Changes 🔄
- Only named export `<Paper />` (no more default export) 🧼
- Moved `<Paper />` out of the `core` folder 🗂️
- Converted Storybook story to CSF 3️⃣
- Added Storybook story for the **outlined** variant of a `<Paper />` 📄
- Added Storybook story for a `<Paper />` that has an `error` 📄

## Preview 📷
Screenshot 2023-07-17 at 12 15 14 PM

## How to test 🧪
- Verify automated testing and type-checking passes
- Check over the app for visual regressions

### CODE REVIEW

File: packages/manager/src/components/Paper.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import { styled } from '@mui/material/styles';
+import * as React from 'react';
+
+import FormHelperText from './core/FormHelperText';
+
+const StyledPaper = styled(_Paper)<Props>(({ theme, ...props }) => ({
+  borderColor: props.error ? `#ca0813` : undefined,
+  padding: theme.spacing(3),
+  paddingTop: 17,
+}));
```
Comment (mjac0bs) [inline packages/manager/src/components/Paper.tsx:11]: Can we move this `StyledPaper` to the bottom of the file to maintain consistency with our docs?

File: packages/manager/src/components/Paper.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import _Paper, { PaperProps } from '@mui/material/Paper';
+import { styled } from '@mui/material/styles';
+import * as React from 'react';
+
+import FormHelperText from './core/FormHelperText';
+
+const StyledPaper = styled(_Paper)<Props>(({ theme, ...props }) => ({
+  borderColor: props.error ? `#ca0813` : undefined,
```
Comment (mjac0bs) [inline packages/manager/src/components/Paper.tsx:8]:
```suggestion
  borderColor: props.error ? theme.color.red : undefined,
```

Can we avoid a hardcoded value for the color here?

File: packages/manager/src/components/Paper.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ */
+export const Paper = (props: Props) => {
+  const { className, error, ...rest } = props;
+
+  return (
+    <React.Fragment>
+      <StyledPaper className={className} {...rest} />
+      {error && (
+        <FormHelperText sx={{ color: '#ca0813' }}>{error}</FormHelperText>
```
Comment (mjac0bs) [inline packages/manager/src/components/Paper.tsx:38]: Same here with the theme color over #ca0813?

**Review by mjac0bs — APPROVED** [review]
Paper looks good throughout the app at various screen sizes. I'm not convinced the `square` prop in Storybook does anything at all, but I also don't see anywhere we're using it, and it's from MUI.

Left a couple small suggestions.

**Comment by bnussman-akamai** [comment]
Great feedback @mjac0bs, thank you. 

The Paper component actually had some existing flaws. Your feedback fixes the dark mode red color. It also fixes the outline behavior.

| Before | After |
|--------|-------|
|   Screenshot 2023-07-17 at 3 47 24 PM     |    Screenshot 2023-07-17 at 3 53 38 PM   |

**Review by cpathipa — APPROVED** [review]

**Pull request merged**